cmd/probe/internal/binding/base.go

```diff
@@ -412,3 +415,83 @@ func (ops *BaseOperations) SwitchoverOps(ctx context.Context, req *bindings.Invo
 	opsRes["event"] = OperationSuccess
 	return opsRes, nil
 }
+
+// JoinMemberOps is used to join the current member into the DB cluster.
+// If OpsResult["event"] == "success" and err == nil, it indicates that the member has successfully left.
+// In any other situation, it signifies a failure.
+func (ops *BaseOperations) JoinMemberOps(ctx context.Context, req *bindings.InvokeRequest, resp *bindings.InvokeResponse) (OpsResult, error) {
+	opsRes := OpsResult{}
+	manager, err := component.GetDefaultManager()
+	if err != nil {
+		opsRes["event"] = OperationFailed
+		opsRes["message"] = err.Error()
+		return opsRes, nil
+	}
+
+	dcsStore := dcs.GetStore()
+	var cluster *dcs.Cluster
+	cluster, err = dcsStore.GetCluster()
+	if err != nil {
+		opsRes["event"] = OperationFailed
+		opsRes["message"] = fmt.Sprintf("get cluster from dcs failed: %v", err)
+		return opsRes, err
```

Comment:
why return error here?

Reply:
To obtain the leader's connection address, the joinMember function relies on the cluster information. If there is a failure in retrieving the cluster information, joinMember will return an error to indicate the failure of the operation.

Reply:
1. why is there no error returned when getting manager fails?